pkg/network/protocols/http2/telemetry.go

```diff
@@ -57,18 +60,45 @@ func newHTTP2KernelTelemetry() *kernelTelemetry {
 
 // update updates the kernel metrics with the given telemetry.
 func (t *kernelTelemetry) update(tel *HTTP2Telemetry) {
-	t.http2requests.Add(int64(tel.Request_seen))
-	t.http2responses.Add(int64(tel.Response_seen))
-	t.endOfStream.Add(int64(tel.End_of_stream))
-	t.endOfStreamRST.Add(int64(tel.End_of_stream_rst))
-	t.pathExceedsFrame.Add(int64(tel.Path_exceeds_frame))
-	t.exceedingMaxInterestingFrames.Add(int64(tel.Exceeding_max_interesting_frames))
-	t.exceedingMaxFramesToFilter.Add(int64(tel.Exceeding_max_frames_to_filter))
+	// We should only add the delta between the current eBPF map state and the last seen eBPF map state
+	telemetryDelta := tel.Sub(t.telemetryLastState)
+	t.http2requests.Add(int64(telemetryDelta.Request_seen))
+	t.http2responses.Add(int64(telemetryDelta.Response_seen))
+	t.endOfStream.Add(int64(telemetryDelta.End_of_stream))
+	t.endOfStreamRST.Add(int64(telemetryDelta.End_of_stream_rst))
+	t.pathExceedsFrame.Add(int64(telemetryDelta.Path_exceeds_frame))
+	t.exceedingMaxInterestingFrames.Add(int64(telemetryDelta.Exceeding_max_interesting_frames))
+	t.exceedingMaxFramesToFilter.Add(int64(telemetryDelta.Exceeding_max_frames_to_filter))
 	for bucketIndex := range t.pathSizeBucket {
-		t.pathSizeBucket[bucketIndex].Add(int64(tel.Path_size_bucket[bucketIndex]))
+		t.pathSizeBucket[bucketIndex].Add(int64(telemetryDelta.Path_size_bucket[bucketIndex]))
 	}
+	// Create a deep copy of the 'tel' parameter to prevent changes from the outer scope affecting the last state
+	t.telemetryLastState = *tel
 }
 
 func (t *kernelTelemetry) Log() {
 	log.Debugf("http2 kernel telemetry summary: %s", t.metricGroup.Summary())
 }
+
+func (t *HTTP2Telemetry) Sub(other HTTP2Telemetry) *HTTP2Telemetry {
+	return &HTTP2Telemetry{
+		Request_seen:                     t.Request_seen - other.Request_seen,
+		Response_seen:                    t.Response_seen - other.Response_seen,
+		End_of_stream:                    t.End_of_stream - other.End_of_stream,
+		End_of_stream_rst:                t.End_of_stream_rst - other.End_of_stream_rst,
+		Path_exceeds_frame:               t.Path_exceeds_frame - other.Path_exceeds_frame,
+		Exceeding_max_interesting_frames: t.Exceeding_max_interesting_frames - other.Exceeding_max_interesting_frames,
+		Exceeding_max_frames_to_filter:   t.Exceeding_max_frames_to_filter - other.Exceeding_max_frames_to_filter,
+		Path_size_bucket:                 computeDifferences(t.Path_size_bucket, other.Path_size_bucket),
+	}
+}
+
+func computeDifferences(arr1, arr2 [8]uint64) [8]uint64 {
```

Comment:
nit: maybe a less generic name given that this function is targeted specifically at `Path_size_bucket`?